lib/internal/http2/compat.js

```diff
@@ -59,6 +60,8 @@ function assertValidHeader(name, value) {
     err = new ERR_HTTP2_PSEUDOHEADER_NOT_ALLOWED();
   } else if (value === undefined || value === null) {
     err = new ERR_HTTP2_INVALID_HEADER_VALUE(value, name);
+  } else if (name === constants.HTTP2_HEADER_CONNECTION) {
+    connectionHeaderMessageWarn();
```

Comment:
It's a bit more nuanced than this... the `Connection` header is only supported in HTTP/2 if the value is specifically (and only) `'trailers'`. All other values are unsupported. If the value is `'trailers'`, then it will be allowed through.

Reply:
OK, Will add this logic 👍